src/libxtp/threecenter.cc

```diff
@@ -137,12 +137,15 @@ void TCMatrix_gwbse::Initialize(Index basissize, Index mmin, Index mmax,
 void TCMatrix_gwbse::MultiplyRightWithAuxMatrix(const Eigen::MatrixXd& matrix) {
   OpenMP_CUDA gemm;
   gemm.setOperators(_matrix, matrix);
-#pragma omp parallel for schedule(dynamic)
-  for (Index i = 0; i < msize(); i++) {
-    gemm.MultiplyRight(_matrix[i]);
+#pragma omp parallel
+  {
+    Index threadid = OPENMP::getThreadId();
+#pragma omp for schedule(dynamic)
+    for (Index i = 0; i < msize(); i++) {
+      gemm.MultiplyRight(_matrix[i], threadid);
+    }
```

Comment:
@JensWehner  could you please explain this new snippet to me?  I am not very sure how the new `threadid` works

Reply:
For the GPU computing you have to retireve temporary data, which depends on the threadid to avoid race conditions, I can retrieve them inside openmp_cuda, but I thought parsing it in explicitly is better, because

a) It makes it more obvious that this method should be used inside an openmp region.
b) there is a small performance benefit to not having to call OPENMP::getThreadId() each time, but instead just pass it in as a function parameter.

Reply:
In this snippet, we first setup the threadpool with `omp parallel` and retrieve the thread id for each thread and then schedule the threads, where inside `gemm` the distribution to GPU and CPU is handled via the `threadid`

Reply:
Could you add that very nice explanation somewhere in the code :) ?